src/livestatus.c

```diff
@@ -0,0 +1,468 @@
+/**
+ * collectd - src/livestatus.c
+ *
+ * Copyright (C) 2019 IN2P3 Computing Centre, IN2P3, CNRS
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ * Authors:
+ *   Remi Ferrand <remi.ferrand at cc.in2p3.fr>
+ **/
+
+#include "collectd.h"
+#include "plugin.h"
+#include "utils/common/common.h"
+
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <sys/un.h>
+
+#define IO_BUFFER_SIZE 4096
+#define LIVESTATUS_FIELD_SEP ';'
+#define LIVESTATUS_PLUGIN_NAME "livestatus"
+
+typedef uint64_t lcounter_t;
```

Comment:
What the benefit of defining a new type over using `derive_t`?

(Note: `derive_t` is a signed `int64_t`, so the below `strtoull` calls would need to become `strtoll` instead.)